src/utils/validator-extras.ts

```diff
@@ -0,0 +1,133 @@
+import dayjs from 'dayjs';
+import forEach from 'lodash/forEach';
+import origValidator from 'validator';
+import type { Attributes, Model } from '../model.js';
+
+type OrigValidator = typeof origValidator;
+
+// extend validator methods
+export interface Extensions {
+  extend(name: string, fn: (...args: unknown[]) => unknown): unknown;
+  // Blocked by: https://github.com/microsoft/TypeScript/issues/7765
+  // NEED_SUGGESTION: How to properly create a custom validation?
+  [name: string]: unknown;
+}
+
+export interface Validator extends OrigValidator {
+  notEmpty(str: string): boolean;
+  len(str: string, min: number, max: number): boolean;
+  isUrl(str: string): boolean;
+  isIPv6(str: string): boolean;
+  isIPv4(str: string): boolean;
+  notIn(str: string, values: string[]): boolean;
+  regex(str: string, pattern: string | RegExp, modifiers: string): boolean;
+  notRegex(str: string, pattern: string, modifiers: string): boolean;
+  min(str: string, val: number): boolean;
+  max(str: string, val: number): boolean;
+  not(str: string, pattern: string, modifiers: string): boolean;
+  notContains(str: string, elem: string): boolean;
+  is(str: string, pattern: string, modifiers: string): boolean;
+  isImmutable<M extends Model>(
+    value: unknown,
+    validatorArgs: unknown[],
+    field: keyof Attributes<M>,
+    modelInstance: Model<any>
+  ): boolean;
+  isNull: OrigValidator['isEmpty'];
+  notNull(val: unknown): boolean;
+  // Blocked by: https://github.com/microsoft/TypeScript/issues/7765
+  // NEED_SUGGESTION: How to properly implement the custom validations?
+  [name: string]: unknown;
+}
+
+const validator: Validator = {
+  ...origValidator,
+  notEmpty(str) {
```

Comment:
We don't need the `Validator` interface, nor `Extensions`, the type can just be added to these methods directly